src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -74,31 +80,37 @@ bool emberAfThermostatClusterAddThermostatSuggestionCallback(CommandHandler * co
 
     // If the effective time in UTC is greater than current time in UTC plus 24 hours, return INVALID_COMMAND.
     const uint32_t kSecondsInDay = 24 * 60 * 60;
-    if (!commandData.effectiveTime.IsNull() && (commandData.effectiveTime.Value() > currentTimestamp + kSecondsInDay))
+    if (!commandData.effectiveTime.IsNull() &&
+        (commandData.effectiveTime.Value() > currentMatterEpochTimestampInSeconds + kSecondsInDay))
     {
         commandObj->AddStatus(commandPath, Status::InvalidCommand);
         return true;
     }
 
     uint8_t uniqueID = delegate->GetUniqueID();
-    ThermostatSuggestionStructWithOwnedMembers thermostatSuggestion;
-    thermostatSuggestion.SetUniqueID(uniqueID);
-    thermostatSuggestion.SetPresetHandle(commandData.presetHandle);
 
-    uint32_t effectiveTime = commandData.effectiveTime.ValueOr(currentTimestamp);
-    thermostatSuggestion.SetEffectiveTime(effectiveTime);
+    Structs::ThermostatSuggestionStruct::Type thermostatSuggestion;
+    thermostatSuggestion.uniqueID     = uniqueID;
+    thermostatSuggestion.presetHandle = commandData.presetHandle;
 
-    const uint32_t kSecondsInMinute = 60;
-    thermostatSuggestion.SetExpirationTime(effectiveTime + (commandData.expirationInMinutes * kSecondsInMinute));
+    uint32_t effectiveTime             = commandData.effectiveTime.ValueOr(currentMatterEpochTimestampInSeconds);
+    thermostatSuggestion.effectiveTime = effectiveTime;
+
+    const uint32_t kSecondsInMinute     = 60;
+    thermostatSuggestion.expirationTime = effectiveTime + (commandData.expirationInMinutes * kSecondsInMinute);
 
     CHIP_ERROR err = delegate->AppendToThermostatSuggestionsList(thermostatSuggestion);
 
-    if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+    if (err != CHIP_NO_ERROR)
     {
         ChipLogError(Zcl, "Failed to AppendToThermostatSuggestionsList with error: %" CHIP_ERROR_FORMAT, err.Format());
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
```

Comment:
Why is this InvalidCommand?  What was invalid about the command?  This does not look like the right error code.